src/NCalc/Expression.cs

```diff
@@ -240,7 +242,17 @@ public Func<TResult> ToLambda<TResult>()
                 ParsedExpression = Compile(OriginalExpression, (Options & EvaluateOptions.NoCache) == EvaluateOptions.NoCache);
             }
 
-            var visitor = new LambdaExpressionVistor(Parameters, Options);
+            LambdaExpressionVistor visitor;
+            System.Linq.Expressions.ParameterExpression parameter = null;
+            if (typeof(TContext) != typeof(Void))
+            {
+                parameter = System.Linq.Expressions.Expression.Parameter(typeof(TContext), "ctx");

```

Comment:
This is just a suggestion since much of this was already there.  System.Linq.Expressions is repeated a fair amount due to the collision with the ncalc Expression.  There's a convention in LamdaExpressionVisitor to use a namespace alias.

```
using L = System.Linq.Expressions;
L.Expression.Parameter(typeof(TContext), "ctx");
```